packages/js/src/dashboard/widgets/search-ranking-compare-widget.js

```diff
@@ -0,0 +1,118 @@
+import { useCallback, useMemo } from "@wordpress/element";
+import { useRemoteData } from "../services/use-remote-data";
+import { Widget } from "./widget";
+import { getDifference } from "../transformers/difference";
+import { SearchRankingCompareWidgetContent } from "./search-ranking-compare/search-ranking-compare-widget-content";
+/**
+ * @typedef {Object<TimeFrame, TimeFrameData>} TimeBasedData The time based data.
+ */
+
+/**
+ * @typedef { "impressions"|"clicks"|"ctr"|"position" } SearchRankingCompareMetric The widget type.
+ */
+
+/**
+ * @typedef {Object} MetricData
+ * @property {number} value - The value of the metric.
+ * @property {number} delta - The delta of the metric.
+ */
+
+/**
+ * @typedef {Object<SearchRankingCompareMetric, MetricData | null>} SearchRankingCompareData The search ranking compare data.
```

Comment:
Instead of allowing any of them to be `null`, shouldn't you specify it below in the ones that are actually optional?
That is actually a plus of you writing the specific properties out below.